camel/toolkits/web_toolkit.py

```diff
@@ -0,0 +1,321 @@
+    “Extract everything” with a broad instruction + broad schema.
+    “Filter locally in JS,” if needed, to pick the relevant link.
+    “Goto or click” to expand or open that detail.
+    “Extract again” with a narrower instruction + schema.
+    “Print final result.”
+    Keep your snippet’s instructions short and direct. Provide one action per act(). For extractions, use one extraction call for each chunk.
+
+    Remember:
+
+    Use observe() to see potential clickable items or possible actions.
+    Use extract() with a carefully chosen instruction and schema to gather data.
+    If you need more data, do another extraction.
+    - Incorporate feedback from previous iterations to improve the plan.
+    Based on this high level task: "{high_level_task}", generate a Stagehand JavaScript snippet with step-by-step instructions.
+    
+    - IMPORTANT: 
+    1. You are a Low-Level Planner that writes a Stagehand JavaScript snippet.  
+        Remember to produce the final result as a JSON object called 'updated_state', which the system will read as:
+
+        {{
+        "status": "success",
+        "updated_state": {{
+            "title": ...,
+            "finalAnswer": ...,
+            "uniqueItems": [...],
+            "link": ...
+        }}
+        }}
+
+        The the calling agent will provide you feedback on what to inlcude in the 'updated_state'. At the end of your snippet, always do the final extraction to fill these fields. For example:
+
+        const updated_state = {{
+        status: "success",
+        updated_state: {{
+            title: extractedTitle,
+            finalAnswer: extractedFinalAnswer,
+            uniqueItems: extractedUniqueItems,
+            link: extractedLink
+        }}
+        }};
+        
+    2. Print or log the final data in a JSON-friendly format so the pipeline can read it. For example:
+    console.log("Final updated_state:", updated_state);
+
+    3. If you cannot find the necessary info after multiple steps, log "No relevant data found. Attempt an alternative approach or refine the search."
+
+    4. Keep your snippet concise.
+
+    **Examples of valid atomic instructions** (one per line):
+    await page.goto("https://www.example.com"); await page.act({{ action: "Click the Sign In button." }}); const data = await page.extract({{ instruction: "Extract all text on page.", schema: z.object({{ text: z.string() }}) }});
+
+    **Do not** wrap multiple steps into a single act call. For instance, don’t do:
+    await page.act({{ action: "Click the sign in button and fill the form." }});
+
+    That should be two lines: one for the click, one for the fill.
+
+    Please produce the Stagehand JavaScript snippet now, following all of the above guidelines, always ending with the final extraction snippet for `updated_state`.
+    """
+
+        response = agent.step(BaseMessage.make_user_message("User", stagehand_prompt))
```

Comment:
`BaseMessage. make_user_message` maybe depreciated soon. We can just use raw `string` here. c.c. @Wendong-Fan we neeed to make a plan to depreciate `make_user_message` and `make_assistant_message`.